Can't install Chainer with CUDA on Mac

I tried to install Chainer with CUDA on Mac, but following errors were occurred.
How do I deal with followings, and install Chainer with CUDA?

[Environment]
- HW: MacBook Pro(Mid 2010) with NVIDIA GeForce 320M 256MB
- OS: OS X El Capitan(10.11.6)
- CUDA: CUDA 7.5
- Python: Python 2.7.12 |Anaconda 4.1.1 (x86_64)
- .bash_profile
  - export PATH="/Developer/NVIDIA/CUDA-7.5/bin:$PATH"                                                                                                                                                     
  - export LD_LIBRARY_PATH="/Developer/NVIDIA/CUDA-7.5/lib:/usr/local/cuda/lib/:$LD_LIBRARY_PATH"                                                                                                          
  - export CPATH="/Developer/NVIDIA/CUDA-7.5/include:$CPATH"                                                                                                                                               
  - export CUDA_INC_DIR="/Developer/NVIDIA/CUDA-7.5/bin:$CUDA_INC_DIR"                                                                                                                                     
  - export DYLD_LIBRARY_PATH="/usr/local/cuda/lib:$DYLD_LIBRARY_PATH"                                                                                                                               
- Chainer:1.16.0

[Error Message]
> pip install chainer --no-cache-dir -vvvv

    gcc -fno-strict-aliasing -I/Users/****/.pyenv/versions/anaconda2-4.0.0/envs/chainer/include -arch x86_64 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/cuda/include -I/Users/****/.pyenv/versions/anaconda2-4.0.0/envs/chainer/include/python2.7 -c cupy/cuda/cublas.cpp -o build/temp.macosx-10.6-x86_64-2.7/cupy/cuda/cublas.o
    In file included from cupy/cuda/cublas.cpp:272:
    cupy/cuda/cupy_cuda.h:18:26: error: typedef redefinition with different types ('cublasDataType_t' vs 'enum cudaDataType_t')
    typedef cublasDataType_t cudaDataType;
                             ^
    /usr/local/cuda/include/library_types.h:66:3: note: previous definition is here
    } cudaDataType;
      ^
    cupy/cuda/cublas.cpp:5650:20: error: no matching function for call to 'cublasSgemmEx'
      __pyx_v_status = cublasSgemmEx(((cublasHandle_t)__pyx_v_handle), ((cublasOperation_t)__pyx_v_transa), ((cublasOperation_t)__pyx_v_transb), __pyx_v_m, __pyx_v_n, __pyx_v_k, (&__pyx_v_alpha), ((void const *)__pyx_v_A), ((cudaDataType)__pyx_v_Atype), __pyx_v_lda, ((void const *)__pyx_v_B), ((cudaDataType)__pyx_v_Btype), __pyx_v_ldb, (&__pyx_v_beta), ((void *)__pyx_v_C), ((cudaDataType)__pyx_v_Ctype), __pyx_v_ldc);
                       ^~~~~~~~~~~~~
    /usr/local/cuda/include/cublas_api.h:1587:39: note: candidate function not viable: no known conversion from 'cudaDataType' (aka 'cudaDataType_t') to 'cublasDataType_t' for 9th argument
    CUBLASAPI cublasStatus_t CUBLASWINAPI cublasSgemmEx  (cublasHandle_t handle,
                                          ^
    2 errors generated.
    error: command 'gcc' failed with exit status 1

Well, I sometimes use macOS also to develop neural networks, aside from training in parallel with large gpu system on Ubuntu computer.
Using [external GPU](https://bizon-tech.com/us/bizonbox2-egpu.html/), I can use Chainer v1.14.0 + Titan X + CUDA 7.5.26 + Macbook Pro 13-inch Late 2013 + OS X Yosemite.

![image](https://cloud.githubusercontent.com/assets/4310419/19836036/b51cc3a6-9ed8-11e6-8e35-15236377040c.png)

But with Chainer v1.17.0, I have same error as @bkvogel reports.

```
In file included from cupy/cuda/cublas.cpp:414:
cupy/cuda/cupy_cuda.h:20:26: error: typedef redefinition with different types ('cublasDataType_t' vs 'enum cudaDataType_t')
typedef cublasDataType_t cudaDataType;
                         ^
/usr/local/cuda/include/library_types.h:66:3: note: previous definition is here
} cudaDataType;
  ^
cupy/cuda/cublas.cpp:6430:26: error: no matching function for call to 'cublasSgemmEx'
        __pyx_v_status = cublasSgemmEx(((cublasHandle_t)__pyx_v_handle), ((cublasOperation_t)__pyx_v_transa), ((cublasOperation_t)__pyx_v_transb), __pyx_v_m, __pyx_v_n, __pyx_v_k, (&__pyx_v_alpha), ((void const *)__pyx_v_A), ((cudaDataType)__pyx_v_Atype), __pyx_v_lda, ((void const *)__pyx_v_B), ((cudaDataType)__pyx_v_Btype), __pyx_v_ldb, (&__pyx_v_beta), ((void *)__pyx_v_C), ((cudaDataType)__pyx_v_Ctype), __pyx_v_ldc);
                         ^~~~~~~~~~~~~
/usr/local/cuda/include/cublas_api.h:1587:39: note: candidate function not viable: no known conversion from 'cudaDataType' (aka 'cudaDataType_t') to 'cublasDataType_t' for 9th argument
CUBLASAPI cublasStatus_t CUBLASWINAPI cublasSgemmEx  (cublasHandle_t handle,
                                      ^
2 errors generated.
error: command 'clang' failed with exit status 1
```

@delta2323 
Now I'm working a bit to fix this, so could you please reopen this?

--- a/cupy/cuda/cupy_cuda.h
+++ b/cupy/cuda/cupy_cuda.h
@@ -3,6 +3,12 @@
 #ifndef INCLUDE_GUARD_CUPY_CUDA_H
 #define INCLUDE_GUARD_CUPY_CUDA_H
 
+#ifdef __APPLE__
+// To avoid redefinition error of cudaDataType_t
+// https://github.com/pfnet/chainer/issues/1700
+#define __LIBRARY_TYPES_H__
+#endif
+
 #ifndef CUPY_NO_CUDA
 #include <cublas_v2.h>
 #include <cuda.h>

--- a/cupy/cuda/cupy_cuda.h
+++ b/cupy/cuda/cupy_cuda.h
@@ -3,15 +3,22 @@
 #ifndef INCLUDE_GUARD_CUPY_CUDA_H
 #define INCLUDE_GUARD_CUPY_CUDA_H
 
-#ifdef __APPLE__
-// To avoid redefinition error of cudaDataType_t
-// https://github.com/pfnet/chainer/issues/1700
-#define __LIBRARY_TYPES_H__
+#ifndef CUPY_NO_CUDA
+#include <cuda.h>
 #endif
+
+#ifdef __APPLE__
+#if CUDA_VERSION == 7050
+// To avoid redefinition error of cudaDataType_t
+// caused by including library_types.h.
+// https://github.com/pfnet/chainer/issues/1700
+// https://github.com/pfnet/chainer/pull/1819
+#define __LIBRARY_TYPES_H__
+#endif // #if CUDA_VERSION == 7050
+#endif // #ifdef __APPLE__
 
 #ifndef CUPY_NO_CUDA
 #include <cublas_v2.h>
-#include <cuda.h>
 #include <cuda_profiler_api.h>
 #include <cuda_runtime.h>
 #include <curand.h>

